File: app/src/main/java/com/bestxty/sault/demo/MainActivity.java
package com.bestxty.sault.demo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.bestxty.sault.Sault;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.text_single).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, SingleTaskActivity.class));
            }
        });
        findViewById(R.id.text_multi).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, MutilTaskActivity.class));
            }
        });
    }


    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Sault.getInstance(this).close();
        DownloadApplication.getRefWatcher(this)
                .watch(this);

    }
}

File: app/src/main/java/com/bestxty/sault/demo/MainActivity.kt
package com.bestxty.sault.demo

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.bestxty.sault.Sault
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        text_single.setOnClickListener { startActivity(Intent(this@MainActivity, SingleTaskActivity::class.java)) }
        text_multi.setOnClickListener { startActivity(Intent(this@MainActivity, MutilTaskActivity::class.java)) }
    }


    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }

    override fun onDestroy() {
        super.onDestroy()
        Sault.getInstance(this).close()
        DownloadApplication.getRefWatcher(this)
                .watch(this)

    }
}
